cashu/wallet/api.py

```diff
@@ -0,0 +1,360 @@
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    wallet_name: str = "default"
+    wallet: Wallet = Wallet(
+        url=mint_settings.mint_url,
+        db=str(Path(cashu_settings.cashu_dir) / wallet_name),
+    )
+    active_wallet: bool
+
+
+wallet = WalletModel(active_wallet=True)
+
+
+@app.on_event("startup")
+async def start_wallet():
+    """Starts wallet on startup."""
+    await init_wallet(wallet.wallet)
+    await wallet.wallet.load_mint()
+    logger.info("Wallet started")
+
+
+@app.on_event("shutdown")
+async def shutdown_tor():
+    """Shutdowns wallet on shutdown."""
+    if wallet.wallet.tor.tor_running:
+        logger.info("Stopping Tor")
+        wallet.wallet.tor.stop_daemon()
+    logger.info("Wallet shutdown")
+
+
+@app.get("/health")
+async def health():
+    """Health check"""
+    return {"status": "OK"}
+
+
+@app.post("/lightning/pay")
+async def pay_lightning_invoice(invoice: str):
+    """Pay a lightning invoice."""
+    if not cashu_settings.lightning:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Lightning is not enabled.",
+        )
+    logger.info(f"Wallet status: {wallet.wallet.status()}")
+    amount, fees = await wallet.wallet.get_pay_amount_with_fees(invoice)
+    if amount <= 0:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Amount must be greater than 0.",
+        )
+    if amount + fees > wallet.wallet.available_balance:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Insufficient funds.",
+        )
+    _, send_proofs = await wallet.wallet.split_to_send(wallet.wallet.proofs, amount)
+    await wallet.wallet.pay_lightning(send_proofs, invoice)
+    logger.info(f"Wallet status: {wallet.wallet.status()}")
+    return {"amount": amount, "fees": fees}
+
+
+@app.get("/lightning/invoice")
+async def generate_lightning_invoice(
+    amount: PositiveInt = Query(..., description="Amount to pay"),
+):
+    """Get a lightning invoice."""
+    if not cashu_settings.lightning:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Lightning is not enabled.",
+        )
+    logger.info(f"Wallet status: {wallet.wallet.status()}")
+    invoice = await wallet.wallet.request_mint(amount=amount)
+    if invoice.pr:
+        logger.info(f"Invoice created for amount {amount}")
+        logger.info(f"Invoice {invoice.pr} - invoice hash {invoice.hash}")
+    else:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Error creating invoice.",
+        )
+    return {"invoice": invoice, "amount": amount}
+
+
+@app.get("/lightning/receive")
+async def receive(
+    amount: int = Query(..., gt=0, description="Amount to receive in satoshis"),
+    invoice_hash: str = Query(..., description="Hash of the invoice to be paid"),
```

Comment:
these are not standard names, should be called `description_hash` and also needs `unhashed_description` to work with CLN.

Reply:
updated names, and added `unhashed_description` parameter